CMakeLists.txt

```diff
@@ -101,6 +87,31 @@ option(GINKGO_INSTALL_RPATH "Set the RPATH when installing its libraries." ON)
 option(GINKGO_INSTALL_RPATH_ORIGIN "Add $ORIGIN (Linux) or @loader_path (MacOS) to the installation RPATH." ON)
 option(GINKGO_INSTALL_RPATH_DEPENDENCIES "Add dependencies to the installation RPATH." OFF)
 
+# load executor-specific configuration
+if(GINKGO_BUILD_CUDA)
+    include(cmake/cuda.cmake)
+endif()
+if(GINKGO_BUILD_HIP)
+    include(cmake/hip.cmake)
+endif()
+if(GINKGO_BUILD_OMP)
+    find_package(OpenMP 3.0 REQUIRED)
+endif()
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads REQUIRED)
```

Comment:
We also have core builds only without any backend where technically this would not be required I think? On the other hand, pthread should always be available on Linux I believe. What about Windows?

Reply:
AFAIK on Windows, threads are just built-in and don't need a specific library.